components/Workshop/WorkshopListing.tsx

```diff
@@ -50,10 +61,22 @@ export default function WorkshopListing({
   const toast = useToast();
   const { isOpen, onOpen, onClose } = useDisclosure();
 
-
-  const pastSlots = slots.filter((slot) => isBeforeNow(new Date(`${slot.date}T${slot.end_time}`)));
-  const futureSlots = slots.filter((slot) => !isBeforeNow(new Date(`${slot.date}T${slot.end_time}`)));
-
+  const pastSlots = slots
+    .filter((slot) => isBeforeNow(new Date(`${slot.date}T${slot.end_time}`)))
+    .sort(
+      (a, b) =>
+        new Date(`${b.date}T${b.end_time}`).getTime() -
+        new Date(`${a.date}T${a.end_time}`).getTime()
+    )
+    .slice(0, 5);
+  const futureSlots = slots
+    .filter((slot) => !isBeforeNow(new Date(`${slot.date}T${slot.end_time}`)))
+    .sort(
+      (a, b) =>
+        new Date(`${a.date}T${a.end_time}`).getTime() -
+        new Date(`${b.date}T${b.end_time}`).getTime()
+    )
+    .slice(0, 5);
```

Comment:
Do we only want to show 5 upcoming slots, or all of them?
I would prefer to see all of them so that I can choose a slot that suits me the best. On the other hand, I am happy for us to show all of them once we add paging, see https://github.com/voluntree-uk/app-next/issues/140